### PR TITLE
feat: implements input field

### PR DESCRIPTION
**Pull request by lehmi11**
Issue: https://github.com/ost-cas-fee-adv-23-24/design-system-component-library-football-fanatics/issues/20

### CODE REVIEW

File: components/input-group/InputFieldGroup.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    | 'url'
+    | 'search'
+    | 'file'
+    | 'hidden';
+  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  state: 'default' | 'error' | 'success';
+  required?: boolean;
+  icon?: TIconType;
+  labelText: string;
+  errorMessage?: string;
```
Comment (bladicito) [inline components/input-group/InputFieldGroup.tsx:27]: We could add following properties to the interface: 

- autoCorrect
- autoCapitalize
- spellCheck 

I would add default values of false to each one and make them optional.

File: components/input-group/InputFieldGroup.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  placeholder,
+  as = 'text',
+  state = 'default',
+  required = false,
+  icon,
+  labelText,
+  errorMessage,
+}: IInputFieldGroupProps) => {
+  const [inputText, setInputText] = React.useState(text ? text : '');
+  const inputId = useId();
```
Comment (bladicito) [inline components/input-group/InputFieldGroup.tsx:47]: I did not know that one! I used always the uuid() package :+1:

File: components/input-group/InputFieldGroup.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        <Label size="M" text={labelText} htmlFor={inputId} />
+      </div>
+
+      <div className="relative">
+        <input
+          id={inputId}
+          type={as}
+          placeholder={placeholder}
+          value={inputText}
+          onChange={(event) => setInputText(event.target.value)}
```
Comment (bladicito) [inline components/input-group/InputFieldGroup.tsx:61]: I get the following error on my IDE here: Property  value  does not exist on type  EventTarget & HTMLInputElement but it compiles without problems :unamused:

File: components/typography/Label.tsx [hunk 4]
```diff
@@ -13,11 +13,12 @@ const mapDesign = {
   S: 'text-sm not-italic font-semibold leading-none',
 };
 
-export const Label = ({ text, size }: ILabelProps) => {
+export const Label = ({ text, size, ...labelProps }: ILabelProps) => {
```
Comment (bladicito) [inline components/typography/Label.tsx:16]: I was thinking that the Label component is not part of the typography. I would move it to the components. 

The labels on the Typography section of figma refer to font size/line-height/font-weight definitions to be used on other html tags as h5/h6/p/span/ etc. AS subtitles. 
I would not use Label html tags somewhere else than in conjunction with an input html tag. 

If I would need to use some  Figma label within a text, I would use  for example a h5/h6  followed by a Paragraph with the text. 

What do you think about this?

**Review by bladicito — COMMENTED** [review]
Looks good to me. Thanks for the implementation @lehmi11 !

File: components/input-group/InputFieldGroup.tsx [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+  text,
+  placeholder,
+  as = 'text',
+  state = 'default',
+  required = false,
+  icon,
+  labelText,
+  errorMessage,
+}: IInputFieldGroupProps) => {
+  const [inputText, setInputText] = React.useState(text ? text : '');
```
Comment (bladicito) [inline components/input-group/InputFieldGroup.tsx:46]: I would say that we would need to control the text/state/OnChange/errorMessage in the parent component. After the presentation or at the time of the integration into the next app. :+1:

**Review by bladicito — APPROVED** [review]
Thanks @lehmi11. Looks good to me!

File: components/input-group/InputFieldGroup.tsx (hunk 1 above)
Comment (lehmi11) [reply components/input-group/InputFieldGroup.tsx:27]: Are those functions or booleans and where would they be implemented?

File: components/input-group/InputFieldGroup.tsx (hunk 3 above)
Comment (lehmi11) [reply components/input-group/InputFieldGroup.tsx:61]: My IDE doesn't say anything 🤷‍♂️

File: components/typography/Label.tsx (hunk 4 above)
Comment (lehmi11) [reply components/typography/Label.tsx:16]: Good point. I move Label component to /components

File: components/input-group/InputFieldGroup.tsx (hunk 5 above)
Comment (lehmi11) [reply components/input-group/InputFieldGroup.tsx:46]: Yes, might do it in the parent. We will see in the web app how we should use the inputFieldGroup component and change it accordingly

File: components/input-group/InputFieldGroup.tsx (hunk 1 above)
Comment (bladicito) [reply components/input-group/InputFieldGroup.tsx:27]: yes, booleans on the input tag itself. most of them with an effect for WebKit browsers (safari).

Comment (lehmi11) [reply components/input-group/InputFieldGroup.tsx:27]: Thx for the hint. I didn't know these attributes before. I have implemented them now 😊

**Pull request merged**